opencivicdata/elections/models/candidacy.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Candidacy-related models.
+"""
+from __future__ import unicode_literals
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+from opencivicdata.core.models.base import (
+    OCDBase,
+    LinkBase,
+    OCDIDField,
+)
+from opencivicdata.core.models import (
+    Person,
+    Post,
+    Organization,
+)
+
+
+@python_2_unicode_compatible
+class Candidacy(OCDBase):
+    """
+    A person competing in an election contest to hold a specific office for a term.
+    """
+    id = OCDIDField(
+        ocd_type='candidacy',
+        help_text='Open Civic Data-style id in the format ``ocd-candidacy/{{uuid}}``.',
+    )
+    person = models.ForeignKey(
+        Person,
+        related_name='candidacies',
+        help_text='Reference to the Person who is the candidate.',
+    )
+    post = models.ForeignKey(
+        Post,
+        related_name='candidacies',
+        help_text='Reference to Post represents the public office for which '
+                  'the candidate is competing.',
+    )
+    contest = models.ForeignKey(
+        'elections.CandidateContest',
+        related_name='candidacies',
+        help_text="Reference to an OCD CandidateContest representing the contest "
+                  "in which the candidate is competing.",
+    )
+    candidate_name = models.CharField(
+        max_length=300,
+        help_text="For preserving the candidate's name as it was of the candidacy."
+    )
+    filed_date = models.DateField(
```

Comment:
In our case, all of these dates are exact. I think the source for this data in most states will be a form the candidate files with an election authority. I think you'll either have the full date or no date at all.

But maybe we want to allow users to put in their best guess for the date? Or if the default for OCD is to use fuzzy dates unless there's a good reason to be exact, we would be fine with that.